wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -20,32 +21,46 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
   /////////////////////
   // Public facing API
 
+  def getNumberOfConfirmations(
+      blockHashOpt: Option[DoubleSha256DigestBE]): Future[Int] =
+    blockHashOpt match {
+      case None => Future.successful(0)
```

Comment:
I think the public facing API should change here to be `Future[Option[Int]]` to indicate the block might not be found.